build.zig

```diff
@@ -978,3 +970,70 @@ fn client_docs(
 
     maybe_execute(b, allocator, client_docs_build, "client_docs");
 }
+
+// Executes the JNI tests.
+// Requires the Java SDK installed and JAVA_HOME environment variable set.
+fn jni_tests_run(
+    b: *std.build.Builder,
+    mode: Mode,
+) void {
+    var test_step = b.step("test:jni", "Run the JNI tests");
+    const jni_abi = if (builtin.os.tag == .linux) b.option(
+        enum { gnu, musl },
+        "jni-test-abi",
+        "Which ABI the JVM will run the test uses",
+    ) orelse .gnu else {};
+
+    const java_home = b.env_map.get("JAVA_HOME") orelse {
```

Comment:
I think we might not want this to be part of `./scripts/build.sh java_client`, just to keep the dependency of JDK not a requirement for building the client library.

Although it's a good idea to include `./scripts/build.sh test:jni` as a step inside Maven, so `mvn package` will run all the tests, both from CI and from the developer machine.

Reply:
So, what's the CI job that tests this?

Reply:
It just landed.
Since Maven knows `JAVA_HOME`, it's called from `mvn` build. 

I've made changes on `build.zig` to detect if we need to run the test using musl, it's not a build argument anymore.

```
❯ mvn package
[INFO] Scanning for projects...
[INFO] Inspecting build with total of 1 modules...
[INFO] Installing Nexus Staging features:
[INFO]   ... total of 1 executions of maven-deploy-plugin replaced with nexus-staging-maven-plugin
[INFO] 
[INFO] ------------------< com.tigerbeetle:tigerbeetle-java >------------------
[INFO] Building TigerBeetle Java client 0.0.1-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- exec-maven-plugin:3.1.0:exec (zig-jni-test) @ tigerbeetle-java ---
1/33 test "JNI: check jvm"... OK
2/33 test "JNI: GetVersion"... OK
3/33 test "JNI: FindClass"... OK
4/33 test "JNI: GetSuperclass"... OK
5/33 test "JNI: IsAssignableFrom"... OK
6/33 test "JNI: GetModule"... OK
7/33 test "JNI: Throw"... OK
8/33 test "JNI: ThrowNew"... OK
9/33 test "JNI: ExceptionOccurred"... OK
10/33 test "JNI: ExceptionDescribe"... Exception in thread "main" java.lang.Exception: EXCEPTION DESCRIBED CORRECTLY
OK
11/33 test "JNI: ExceptionClear"... OK
12/33 test "JNI: ExceptionCheck"... OK
13/33 test "JNI: References"... OK
14/33 test "JNI: LocalFrame"... OK
15/33 test "JNI: AllocObject"... OK
16/33 test "JNI: NewObject"... OK
17/33 test "JNI: IsInstanceOf"... OK
18/33 test "JNI: GetFieldId"... OK
19/33 test "JNI: Get<Type>Field, Set<Type>Field"... OK
20/33 test "JNI: GetMethodId"... OK
21/33 test "JNI: Call<Type>Method,CallNonVirtual<Type>Method"... OK
22/33 test "JNI: GetStaticFieldId"... OK
23/33 test "JNI: GetStatic<Type>Field, SetStatic<Type>Field"... OK
24/33 test "JNI: GetStaticMethodId"... OK
25/33 test "JNI: CallStatic<Type>Method"... OK
26/33 test "JNI: strings"... OK
27/33 test "JNI: strings utf"... OK
28/33 test "JNI: GetStringRegion"... OK
29/33 test "JNI: GetStringUTFRegion"... OK
30/33 test "JNI: GetStringCritical"... OK
31/33 test "JNI: DirectByteBuffer"... OK
32/33 test "JNI: object array"... OK
33/33 test "JNI: primitive arrays"... OK
All 33 tests passed.
[INFO] 
[INFO] --- exec-maven-plugin:3.1.0:exec (zig-build) @ tigerbeetle-java ---
```